intercom-java/src/main/java/io/intercom/api/Job.java

```diff
@@ -0,0 +1,216 @@
+package io.intercom.api;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JavaType;
+import com.google.common.collect.Maps;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@SuppressWarnings("UnusedDeclaration")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+public class Job extends TypedData {
+
+    private static final HashMap<String, String> SENTINEL = Maps.newHashMap();
+
+    public static Job find(String id)
+        throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return find(id, SENTINEL);
+    }
+
+    public static Job find(String id, Map<String, String> params)
+        throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return new HttpClient(
+            UriBuilder.newBuilder().path("jobs").path(id).query(params).build()
+        ).get(Job.class);
+    }
+
+    public static Job update(Job job)
+        throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return DataResource.post(
+            job,
+            UriBuilder.newBuilder().path("jobs").path(job.getID()).build(),
+            Job.class);
+    }
+
+    static <T extends TypedData> Job submit(List<JobItem<T>> items, List<String> bulkPaths)
+        throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return submit(items, null, bulkPaths);
+    }
+
+    static <T extends TypedData> Job submit(List<JobItem<T>> items, Job job, List<String> bulkPaths)
+        throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return DataResource.create(new JobItemRequest<T>(items, job), bulkPaths, Job.class);
+    }
```

Comment:
Job.submit's not exposed directly - constrains the client to only support what's available in the beta bulk api (events,users) and to submit jobs with specific types for now.